youtube_dlc/postprocessor/ffmpeg.py

```diff
@@ -349,21 +376,35 @@ def run(self, information):
 class FFmpegVideoRemuxerPP(FFmpegPostProcessor):
     def __init__(self, downloader=None, preferedformat=None):
         super(FFmpegVideoRemuxerPP, self).__init__(downloader)
-        self._preferedformat = preferedformat
+        self._preferedformats = preferedformat.lower().split('/')
 
     def run(self, information):
         path = information['filepath']
-        if information['ext'] == self._preferedformat:
-            self.to_screen('Not remuxing video file %s - already is in target format %s' % (path, self._preferedformat))
+        sourceext, targetext = information['ext'].lower(), None
+        for pair in self._preferedformats:
+            kv = pair.split('>')
+            if len(kv) == 1 or kv[0].strip() == sourceext:
+                targetext = kv[-1].strip()
+                break
+
+        _skip_msg = (
+            'could not find a mapping for %s' if not targetext
+            else 'already is in target format %s' if sourceext == targetext
+            else None)
+        if _skip_msg:
+            self.to_screen('Not remuxing media file %s - %s' % (path, _skip_msg % sourceext))
             return [], information
+
         options = ['-c', 'copy', '-map', '0', '-dn']
-        prefix, sep, ext = path.rpartition('.')
-        outpath = prefix + sep + self._preferedformat
-        self.to_screen('Remuxing video from %s to %s, Destination: ' % (information['ext'], self._preferedformat) + outpath)
+        if sourceext in ['mp4', 'm4a', 'mov']:
+            options.extend(['-movflags', '+faststart'])
```

Comment:
Looks like this was a mistake on my behalf; the check needs to be performed on `targetext` instead of `sourceext`.

`-movflags +faststart` makes ffmpeg rearrange the moov atom of the mp4 to the start of the file, so such mp4 files can start instantly and can be streamed over a network effortlessly like an MKV file. It requires a full rewrite of the file, however IMO the benefits outweigh needing to wait a few more seconds for the remux to complete

It's only an issue for when youtube-dl outputs to something that can't be seeked (like stdout) but you wouldn't be using -remux in that situation in the first place

https://www.reddit.com/r/ffmpeg/comments/alxhlj/question_on_when_to_use_movflags_faststart/

Reply:
fixed

Reply:
It is completely lossless still, right?

Reply:
@Jules-A yes, it's the equivalent of ffmpeg running `-map 0 -copy c` when doing the second pass